secrets/provider/kubernetes/secrets.go

```diff
@@ -33,15 +33,22 @@ const (
 
 // NewProvider returns a Kubernetes secrets provider.
 func NewProvider() provider.SecretBackendProvider {
-	return k8sProvider{Backend}
+	return k8sProvider{}
 }
 
 type k8sProvider struct {
-	name string
+}
+
+// ValidateConfig implements SecretBackendProvider.
+func (p k8sProvider) ValidateConfig(oldCfg, newCfg provider.ProviderConfig) error {
```

Comment:
do we have a test for this?

Reply:
not yet - was waiting to add schema support in next PR

Reply:
added a quick test for now